### PR TITLE
Fix 14 issues found in ticket backlog

### LINKED ISSUE #487: GameUpdateOverlay crash
### Version

1.11.0_1484 

### Brief Description

Crash when max message length is set to 1 in configs.

### Steps to Reproduce

Set maximum message length to 1 and trigger any game overlay update.

### Crash Report

https://hst.sh/adikejuzel.css

### Installed Mods

_No response_

### LINKED ISSUE #488: QuestBookListPage crash in specific edge cases
### Version

1.11.0_1484 

### Brief Description

IndexOutOfBoundsException in specific edge cases.

### Crash Report

https://hst.sh/qomokanuto

### Installed Mods

_No response_

### LINKED ISSUE #489: Dialogue history does not update in the quest book
### Version

1.11.0_1484 

### Brief Description

Dialogue history does not update in the quest book

### Steps to Reproduce

Talk to NPC, open quest book, dialogue is not updated.

### Expected Behaviour

New dialogue should appear.

### Actual Behaviour

Dialogue history stays the same.

### Installed Mods

_No response_

### LINKED ISSUE #491: Deleting chat tabs does not update ChatGUI
### Version

1.11.0_1484 

### Brief Description

Tab list does not get updated, is only set at init.

### Steps to Reproduce

Delete a tab and do not close Chat. Click on the deleted tab.

### Crash Report

https://hst.sh/ugeyohawin

### Installed Mods

_No response_

### LINKED ISSUE #492: Translation feature bug causes chat text not to show up
### Version

1.11.0

### Brief Description

When using translation, chat messages don't show up.

https://hst.sh/zehovihuvu

image
image
image

### Steps to Reproduce

Enable translations.

### Expected Behaviour

If translation fails, show original text.

### Actual Behaviour

No text is shown.

### Installed Mods

_No response_

### LINKED ISSUE #493: Being in a party breaks objective overlay
### Version

1.11.0_1484 

### Brief Description

Objectives are not getting parsed correctly if player is in party.

### Steps to Reproduce

Join a party.

### Expected Behaviour

Correctly parsed objectives

### Actual Behaviour

Incorrect objective parsing

### Installed Mods

_No response_

### LINKED ISSUE #494: Guild objective does not disappear after completion
### Version

1.11.0_1484 

### Brief Description

Guild objective does not disappear after completion

### Steps to Reproduce

Complete guild objective

### Expected Behaviour

Upon completion, objective disappears from overlay.

### Actual Behaviour

Upon completion, objective won't disappear from the overlay until /class.

### Installed Mods

_No response_

### LINKED ISSUE #496: Race condition in ServerEvents
### Version

1.11.0

### Brief Description

Race condition if player is loading in slowly.

### Steps to Reproduce

https://hst.sh/jehecudogo

### Expected Behaviour

No error.

### Actual Behaviour

Error preventing player from joining.

### Installed Mods

_No response_

### LINKED ISSUE #497: Race condition in ClientEvents prevents joining Wynn
### Version

1.11.0_1484 

### Brief Description

https://hst.sh/siqehemuwa

### Steps to Reproduce

Join Wynn and be lucky enough to experience the race condition.

### Expected Behaviour

No problem joining.

### Actual Behaviour

Exceptions thrown when joining.

### Installed Mods

_No response_

### LINKED ISSUE #498: NPE in ServerEvents
### Version

1.11.0_1484 

### Brief Description

https://hst.sh/okenumexup

### Steps to Reproduce

-

### Expected Behaviour

-

### Actual Behaviour

-

### Installed Mods

_No response_

### PR DESCRIPTION
**Pull request by kristofbolyai**
This PR includes a huge amount of fixes of unrelated things. These are all the issues I found looking through all tickets. Some of these are just null-checks, some are slight reworks of systems. All of these were tested to some level, some crashes were tested by their ticket owners as well.

I know that this PR seems like a mess, but if reviewed commit by commit, it is easy to navigate. I also think that making 15 PRs is not the way to do this.

List of fixes and their issues:
- https://github.com/Wynntils/Wynntils/commit/936fd31c3ffd527216f8a836617f2f18d778ef57 fixes https://github.com/Wynntils/Wynntils/issues/488
- https://github.com/Wynntils/Wynntils/commit/cf2ba8e38f9b79877432a66964be1d4ccad99f11 fixes https://github.com/Wynntils/Wynntils/issues/487
- https://github.com/Wynntils/Wynntils/commit/c0a493008ae873a6a76a59615351282842849dbe fixes https://github.com/Wynntils/Wynntils/issues/491
- https://github.com/Wynntils/Wynntils/commit/96e3b752a472bf63faa9ed97104b540b185424a4 fixes https://github.com/Wynntils/Wynntils/issues/492
- https://github.com/Wynntils/Wynntils/commit/6d28f5542976c1ee1327b1d8524389b19453a20a fixes https://github.com/Wynntils/Wynntils/issues/493
- https://github.com/Wynntils/Wynntils/commit/476808451bc606c3827dfa8ad946a0b77f2554bb fixes https://github.com/Wynntils/Wynntils/issues/494
- https://github.com/Wynntils/Wynntils/commit/37c8f6b9c7bc4acd83e91f14a65249147f96aa37 fixes https://github.com/Wynntils/Wynntils/issues/496
- https://github.com/Wynntils/Wynntils/commit/45e3f1ac9c9e8bbc067149ce7f280c1e54757432 fixes https://github.com/Wynntils/Wynntils/issues/497
- https://github.com/Wynntils/Wynntils/commit/18442637a6d534143564cfbe157b461ce2929cfc fixes https://github.com/Wynntils/Wynntils/issues/498
- https://github.com/Wynntils/Wynntils/commit/1741c0ee0b2868217ece75d83630de312f5bbf9e fixes https://github.com/Wynntils/Wynntils/issues/499
- https://github.com/Wynntils/Wynntils/commit/3fd18d05d6442767b49c3dc3a2e86021481f9055 (this fix does not have an issue on GH)
- https://github.com/Wynntils/Wynntils/commit/c1ec83151e88960a761ecac8826450726fc6c743, https://github.com/Wynntils/Wynntils/pull/502/commits/ad7695457f20002bbdec0979d6f5c614faa5ba2e fixes https://github.com/Wynntils/Wynntils/issues/500
- https://github.com/Wynntils/Wynntils/commit/81d760b4166114e5df406acbd90c752c77acfabf fixes https://github.com/Wynntils/Wynntils/issues/501
- https://github.com/Wynntils/Wynntils/commit/cf0407b443dcd4f9048fd2bef501e3bbf669358e fixes https://github.com/Wynntils/Wynntils/issues/489

### CODE REVIEW

**Review by DevScyu — APPROVED** [review]

File: src/main/java/com/wynntils/webapi/services/CachingTranslationService.java [hunk 1]
```diff
@@ -81,6 +84,10 @@ public static synchronized void loadTranslationCache() {
             translationCaches = gson.fromJson(json, type);
         } catch (IOException e) {
             e.printStackTrace();
+        } finally {
+            if (translationCaches == null) {
```
Comment (magicus) [inline src/main/java/com/wynntils/webapi/services/CachingTranslationService.java:88]: Maybe it's not worth dwelling into, but why does this happen? (This is more of a workaround than a fix (but I'm okay with that, just curious))

File: src/main/java/com/wynntils/modules/utilities/events/ClientEvents.java [hunk 2]
```diff
@@ -162,7 +162,9 @@ public void classChange(WynnClassChangeEvent e) {
         afkProtectionEnabled = false;
         afkProtectionActivated = false;
 
-        lastHealth = McIf.player().getHealth();
+        EntityPlayerSP player = McIf.player();
+        if (player != null)
```
Comment (magicus) [inline src/main/java/com/wynntils/modules/utilities/events/ClientEvents.java]: As usual, I dislike multi-line if statements without braces. Either make it single line, or add braces. (And do a google search for "apple goto fail" for an enlightening story :))

File: src/main/java/com/wynntils/webapi/services/CachingTranslationService.java (hunk 1 above)
Comment (kristofbolyai) [reply src/main/java/com/wynntils/webapi/services/CachingTranslationService.java:88]: I am not sure either, I obviously could not figure out from the logs, however, this should prevent most issues, except that we might lose a cache. Though I still find it annoying that gson did not throw an NPE and just converted null to a "null" json file. I think the ticket mentioned some ancient version working, 1.9.5 iirc, but would have to check. Maybe an answer lies there.

**Review by magicus — APPROVED** [review]
Looks good (bar the multi-line if ;-)). Thanks for fixing all these! Really, really great work! 🎉

File: src/main/java/com/wynntils/webapi/services/CachingTranslationService.java (hunk 1 above)
Comment (magicus) [reply src/main/java/com/wynntils/webapi/services/CachingTranslationService.java:88]: Probably not worth spending more time on it then.

**Pull request merged**